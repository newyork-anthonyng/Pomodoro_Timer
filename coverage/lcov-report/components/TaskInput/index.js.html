<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/TaskInput/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/TaskInput</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { PureComponent } from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
&nbsp;
class TaskInput extends PureComponent {
  constructor() {
    super();
&nbsp;
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }
&nbsp;
  handleKeyPress(e) {
    <span class="missing-if-branch" title="else path not taken" >E</span>if (e.key === 'Enter') {
      this.props.onSubmit();
    }
  }
&nbsp;
  render() {
    const { value, onChange, onSubmit } = this.props;
&nbsp;
    return (
      &lt;div&gt;
        &lt;textarea
          value={value}
          onChange={onChange}
          onKeyPress={this.handleKeyPress}
        /&gt;
        &lt;button onClick={onSubmit}&gt;âœ“&lt;/button&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
TaskInput.defaultProps = {
  onSubmit: noop,
  onChange: noop,
};
&nbsp;
TaskInput.propTypes = {
  value: T.string.isRequired,
  onSubmit: T.func.isRequired,
  onChange: T.func.isRequired,
};
&nbsp;
export default TaskInput;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 05 2017 22:18:16 GMT-0700 (PDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
