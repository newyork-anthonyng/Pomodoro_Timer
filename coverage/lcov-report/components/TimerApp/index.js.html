<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/TimerApp/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/TimerApp</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import TimeDisplay from '../TimeDisplay';
import Button from '../Button';
import TaskInput from '../TaskInput';
import TaskList from '../TaskList';
&nbsp;
const TimerApp = ({
  onPlayClick,
  onPauseClick,
  onResetClick,
  timeInMs,
  timerIsActive,
  tasks,
  onTaskInputChange,
  taskInputValue,
  onTaskSubmit,
  onTaskDeleteClick,
}) =&gt; (
  &lt;div&gt;
    &lt;TimeDisplay timeInMs={timeInMs} /&gt;
&nbsp;
    &lt;Button
      text={timerIsActive ? 'Pause' : 'Play'}
      onClick={timerIsActive ? onPauseClick : onPlayClick}
    /&gt;
&nbsp;
    &lt;Button
      text="Reset"
      onClick={onResetClick}
    /&gt;
&nbsp;
    &lt;TaskInput
      value={taskInputValue}
      onSubmit={onTaskSubmit}
      onChange={onTaskInputChange}
    /&gt;
&nbsp;
    &lt;TaskList
      tasks={tasks}
      onDeleteClick={onTaskDeleteClick}
    /&gt;
&nbsp;
  &lt;/div&gt;
);
&nbsp;
TimerApp.defaultProps = {
  onPlayClick: noop,
  onPauseClick: noop,
  onResetClick: noop,
  onTaskInputChange: noop,
  onTaskSubmit: noop,
  onTaskDeleteClick: noop,
};
&nbsp;
TimerApp.propTypes = {
  onPlayClick: T.func.isRequired,
  onPauseClick: T.func.isRequired,
  onResetClick: T.func.isRequired,
  timeInMs: T.number.isRequired,
  timerIsActive: T.bool.isRequired,
  tasks: T.arrayOf(T.shape({
    title: T.string.isRequired,
    text: T.string.isRequired,
  })).isRequired,
  onTaskInputChange: T.func.isRequired,
  taskInputValue: T.string.isRequired,
  onTaskSubmit: T.func.isRequired,
  onTaskDeleteClick: T.func.isRequired,
};
&nbsp;
export default TimerApp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 05 2017 22:18:16 GMT-0700 (PDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
