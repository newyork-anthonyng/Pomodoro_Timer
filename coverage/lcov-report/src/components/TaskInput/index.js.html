<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/TaskInput/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/TaskInput</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">120x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { PureComponent } from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import glamorous from 'glamorous';
&nbsp;
const Container = glamorous.div({
  marginTop: '20px',
});
const Textarea = glamorous.textarea({
  display: 'block',
  fontSize: '16px',
  height: '100px',
  marginBottom: '10px',
  width: '340px',
});
&nbsp;
const TaskInput = ({ value, onChange, onSubmit }) =&gt; (
  &lt;Container&gt;
    &lt;Textarea
      value={value}
      onChange={onChange}
    /&gt;
    &lt;button onClick={onSubmit}&gt;âœ“&lt;/button&gt;
  &lt;/Container&gt;
);
&nbsp;
TaskInput.defaultProps = {
  onSubmit: noop,
  onChange: noop,
};
&nbsp;
TaskInput.propTypes = {
  value: T.string.isRequired,
  onSubmit: T.func.isRequired,
  onChange: T.func.isRequired,
};
&nbsp;
export default TaskInput;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 12 2017 05:46:05 GMT-0700 (PDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
