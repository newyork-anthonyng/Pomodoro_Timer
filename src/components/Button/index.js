import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';

const Button = ({ text, onClick, style }) => (
  <button style={style} onClick={onClick}>{text}</button>
);

Button.defaultProps = {
  onClick: noop,
};

Button.propTypes = {
  text: T.string.isRequired,
  onClick: T.func.isRequired,
  style: T.object,
};

export default Button;
