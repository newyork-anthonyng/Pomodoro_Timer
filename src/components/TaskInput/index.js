import React, { PureComponent } from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import glamorous from 'glamorous';

const Container = glamorous.div({
  marginTop: '20px',
});
const Textarea = glamorous.textarea({
  display: 'block',
  fontSize: '16px',
  height: '100px',
  marginBottom: '10px',
  width: '340px',
});

const TaskInput = ({ value, onChange, onSubmit }) => (
  <Container>
    <Textarea
      value={value}
      onChange={onChange}
    />
    <button onClick={onSubmit}>âœ“</button>
  </Container>
);

TaskInput.defaultProps = {
  onSubmit: noop,
  onChange: noop,
};

TaskInput.propTypes = {
  value: T.string.isRequired,
  onSubmit: T.func.isRequired,
  onChange: T.func.isRequired,
};

export default TaskInput;
