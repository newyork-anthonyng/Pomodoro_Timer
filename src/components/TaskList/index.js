import React from 'react';
import T from 'prop-types';
import Task from '../Task';
import noop from '../../utility/noop';
import glamorous from 'glamorous';

const Container = glamorous.div({
  height: '250px',
  marginTop: '40px',
  overflow: 'scroll',
  width: '340px',
});

const TaskList = ({ tasks, onDeleteClick }) => (
  <Container>
    {
      tasks.map((task) => (
        <Task
          {...task}
          key={task.id}
          onDeleteClick={onDeleteClick}
        />
      ))
    }
  </Container>
);

TaskList.defaultProps = {
  tasks: [],
  onDeleteClick: noop,
};

TaskList.propTypes = {
  tasks: T.arrayOf(T.shape({
    id: T.string.isRequired,
    title: T.string.isRequired,
    text: T.string.isRequired,
  })).isRequired,
  onDeleteClick: T.func.isRequired,
};

export default TaskList;
