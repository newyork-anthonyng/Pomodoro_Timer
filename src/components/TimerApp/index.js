import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import TimeDisplay from '../TimeDisplay';
import Button from '../Button';
import TaskInput from '../TaskInput';
import TaskList from '../TaskList';

const TimerApp = ({
  onPlayClick,
  onPauseClick,
  onResetClick,
  timeInMs,
  timerIsActive,
  tasks,
  onTaskInputChange,
  taskInputValue,
  onTaskSubmit,
  onTaskDeleteClick,
  onClearAllClick,
}) => (
  <div>
    <TimeDisplay timeInMs={timeInMs} />

    <Button
      text={timerIsActive ? 'Pause' : 'Play'}
      onClick={timerIsActive ? onPauseClick : onPlayClick}
      style={{ marginRight: '20px' }}
    />

    <Button
      text="Reset"
      onClick={onResetClick}
    />

    <TaskInput
      value={taskInputValue}
      onSubmit={onTaskSubmit}
      onChange={onTaskInputChange}
    />

    <TaskList
      tasks={tasks}
      onDeleteClick={onTaskDeleteClick}
    />

    <Button
      text="Clear All"
      onClick={onClearAllClick}
    />

  </div>
);

TimerApp.defaultProps = {
  onPlayClick: noop,
  onPauseClick: noop,
  onResetClick: noop,
  onTaskInputChange: noop,
  onTaskSubmit: noop,
  onTaskDeleteClick: noop,
  onClearAllClick: noop,
};

TimerApp.propTypes = {
  onPlayClick: T.func.isRequired,
  onPauseClick: T.func.isRequired,
  onResetClick: T.func.isRequired,
  timeInMs: T.number.isRequired,
  timerIsActive: T.bool.isRequired,
  tasks: T.arrayOf(T.shape({
    title: T.string.isRequired,
    text: T.string.isRequired,
  })).isRequired,
  onTaskInputChange: T.func.isRequired,
  taskInputValue: T.string.isRequired,
  onTaskSubmit: T.func.isRequired,
  onTaskDeleteClick: T.func.isRequired,
  onClearAllClick: T.func.isRequired,
};

export default TimerApp;
