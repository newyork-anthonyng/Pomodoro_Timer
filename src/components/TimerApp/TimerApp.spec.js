import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../Button';
import TaskInput from '../TaskInput';
import TaskList from '../TaskList';
import TimerApp from './';
import noop from '../../utility/noop';

const defaultProps = {
  onPlayClick: noop,
  onPauseClick: noop,
  onResetClick: noop,
  timeInMs: 25 * 60 * 1000, // 25 minutes
  timerIsActive: false,
  tasks: [
    { id: 'batman1', title: 'Batman', text: 'The Dark Knight' },
    { id: 'superman2', title: 'Superman', text: '???' },
  ],
  onTaskInputChange: noop,
  taskInputValue: 'Learn Marko',
  onTaskSubmit: noop,
  onTaskDeleteClick: noop,
};

it('should render correctly when timer is active', () => {
  const wrapper = shallow(
    <TimerApp
      {...defaultProps}
      timerIsActive
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('should render correctly when timer is not active', () => {
  const wrapper = shallow(
    <TimerApp
      {...defaultProps}
      timerIsActive={false}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('should run callback when clicking on Play button', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <TimerApp
      {...defaultProps}
      onPlayClick={cb}
      timerIsActive={false}
    />
  );
  const playButton =  wrapper.find(Button).at(0);
  playButton.simulate('click');

  expect(cb.mock.calls.length).toEqual(1);
});

it('should run callback when clicking on Pause button', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <TimerApp
      {...defaultProps}
      onPauseClick={cb}
      timerIsActive
    />
  );
  const pauseButton = wrapper.find(Button).at(0);
  pauseButton.simulate('click');

  expect(cb.mock.calls.length).toEqual(1);
});

it('should pass correct props to Reset button', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <TimerApp
      {...defaultProps}
      onResetClick={cb}
    />
  );
  const resetButton = wrapper.find(Button).at(1);
  resetButton.simulate('click');

  expect(cb.mock.calls.length).toEqual(1);
});

it('should pass correct props to TaskInput', () => {
  const value = 'Run marathon';
  const onSubmit = jest.fn();
  const onChange = jest.fn();
  const wrapper = shallow(
    <TimerApp
      {...defaultProps}
      taskInputValue={value}
      onTaskInputChange={onChange}
      onTaskSubmit={onSubmit}
    />
  );
  const taskInput = wrapper.find(TaskInput);

  expect(taskInput.props().value).toEqual(value);
  expect(taskInput.props().onSubmit).toBe(onSubmit);
  expect(taskInput.props().onChange).toBe(onChange);
});

it('should pass correct props to TaskList', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <TimerApp
      {...defaultProps}
      onTaskDeleteClick={cb}
    />
  );
  const taskList = wrapper.find(TaskList);

  expect(taskList.props().tasks).toBe(defaultProps.tasks);
  expect(taskList.props().onDeleteClick).toBe(cb);
});
