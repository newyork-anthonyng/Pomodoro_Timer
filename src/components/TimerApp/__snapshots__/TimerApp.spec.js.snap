// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when timer is active 1`] = `
<div>
  <TimeDisplay
    timeInMs={1500000}
  />
  <Button
    onClick={[Function]}
    style={
      Object {
        "marginRight": "20px",
      }
    }
    text="Pause"
  />
  <Button
    onClick={[Function]}
    text="Reset"
  />
  <TaskInput
    onChange={[Function]}
    onSubmit={[Function]}
    value="Learn Marko"
  />
  <TaskList
    onDeleteClick={[Function]}
    tasks={
      Array [
        Object {
          "id": "batman1",
          "text": "The Dark Knight",
          "title": "Batman",
        },
        Object {
          "id": "superman2",
          "text": "???",
          "title": "Superman",
        },
      ]
    }
  />
  <Button
    onClick={[Function]}
    text="Clear All"
  />
</div>
`;

exports[`should render correctly when timer is not active 1`] = `
<div>
  <TimeDisplay
    timeInMs={1500000}
  />
  <Button
    onClick={[Function]}
    style={
      Object {
        "marginRight": "20px",
      }
    }
    text="Play"
  />
  <Button
    onClick={[Function]}
    text="Reset"
  />
  <TaskInput
    onChange={[Function]}
    onSubmit={[Function]}
    value="Learn Marko"
  />
  <TaskList
    onDeleteClick={[Function]}
    tasks={
      Array [
        Object {
          "id": "batman1",
          "text": "The Dark Knight",
          "title": "Batman",
        },
        Object {
          "id": "superman2",
          "text": "???",
          "title": "Superman",
        },
      ]
    }
  />
  <Button
    onClick={[Function]}
    text="Clear All"
  />
</div>
`;
